src/NuGet.Clients/NuGet.CommandLine/GlobalSuppressions.cs

```diff
@@ -53,7 +51,7 @@
 [assembly: SuppressMessage("Build", "CA1062:In externally visible method 'string ProjectFactory.InitializeProperties(IPackageMetadata metadata)', validate parameter 'metadata' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.CommandLine.ProjectFactory.InitializeProperties(NuGet.Packaging.IPackageMetadata)~System.String")]
 [assembly: SuppressMessage("Build", "CA1062:In externally visible method 'IProjectFactory ProjectFactory.ProjectCreator(PackArgs packArgs, string path)', validate parameter 'packArgs' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.CommandLine.ProjectFactory.ProjectCreator(NuGet.Commands.PackArgs,System.String)~NuGet.Commands.IProjectFactory")]
 [assembly: SuppressMessage("Build", "CA1305:The behavior of 'string.Format(string, params object[])' could vary based on the current user's locale settings. Replace this call in 'ProjectFactory.WriteDetail(string, params object[])' with a call to 'string.Format(IFormatProvider, string, params object[])'.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.CommandLine.ProjectFactory.WriteDetail(System.String,System.Object[])")]
-[assembly: SuppressMessage("Build", "CA1801:Parameter registryKeyPath of method GetValueFromRegistryKey is never used. Remove the parameter or use it in the method body.", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.CommandLine.RegistryKeyUtility.GetValueFromRegistryKey(System.String,System.String,Microsoft.Win32.RegistryKey,NuGet.Common.ILogger)~System.Object")]
+[assembly: SuppressMessage("Build", "CA1801:Parameter registryKeyPath of method GetValueFromRegistryKey is never used. Remove the parameter or use it in the method body.", Justification = "Shipped Public API", Scope = "member", Target = "~M:NuGet.CommandLine.RegistryKeyUtility.GetValueFromRegistryKey(System.String,System.String,Microsoft.Win32.RegistryKey,NuGet.Common.ILogger)~System.Object")]
```

Comment:
I no longer remember why I made NuGet.CommandLine use the public API analyzers. Although it's packed, it goes to `tools/nuget.exe` in the package, meaning customers can't install this package in their projects and use its APIs. Hence we shouldn't have any public API concerns with this project.

Reply:
Fixed parameter usage in this case and removed unused params in NuGet.CommandLine.